src/test/java/uk/gov/register/functional/DataDownloadFunctionalTest.java

```diff
@@ -143,23 +143,44 @@ public void downloadPartialRSF_shouldReturnAPartOfRegisterAsRsfStream() throws I
     }
 
     @Test
-    public void downloadPartialRSF_shouldReturnReturn404ForRsfBoundariesInTheIncorrectOrder() throws IOException {
-        Response response = register.getRequest("/download-rsf/4/1");
-        
+    public void downloadPartialRSF_shouldReturn400ForRsfBoundariesOutOfBound() throws IOException {
+        Response response = register.getRequest("/download-rsf/666/1000");
+
         assertThat(response.getStatus(), equalTo(400));
     }
 
     @Test
-    public void downloadPartialRSF_shouldReturnReturn404ForRsfBoundariesOutOfBound() throws IOException {
-        Response response = register.getRequest("/download-rsf/666/1000");
+    public void downloadPartialRSF_shouldReturn400_whenTotalEntries1OutOfBounds() {
+        Response response = register.getRequest("/download-rsf/-1/2");
+
+        assertThat(response.getStatus(), equalTo(400));
+    }
+
+    @Test
+    public void downloadPartialRSF_shouldReturn400_whenTotalEntries2LessThanTotalEntries1() {
+        Response response = register.getRequest("/download-rsf/2/1");
+
+        assertThat(response.getStatus(), equalTo(400));
+    }
+
+    @Test
+    public void downloadPartialRSF_shouldReturn400_whenRequestedTotalEntriesExceedsEntriesInRegister() {
```

Comment:
does this duplicate `downloadPartialRSF_shouldReturn400ForRsfBoundariesOutOfBound`?